### PR TITLE
WhiteRabbit 1.0, with Snowflake support

### PR DESCRIPTION
**Pull request by janblom**
This is a very large pull request due to many refactorings that were needed to be able to isolate the database facing code, which contributes to modularity, simpler code, better testability and easier support for new (JDBC) databases. I do not recommend to review all code changes, but if you do want to review, start from the newly added interface StorageHandler and class Snowflake handler.

So, new in this release:

- Snowflake support
- integration tests for PostgreSQL, Oracle, MySQL, CSV, SAS (run with `mvn verify`). These tests depend on having Docker available. There is also an integration test for Snowflake, but that only runs when credentials are supplied (see README.md)
- updated dependencies as far as possible without code changes 
- the distributable packages run with Java 1.8 and higher, but the tests can only be run with Java 17. Maven has been configured to fail if the available Java version is not 17.

One caveat: all tests that involve the GUI (Swing) fail on MacOS with UnspportedOperationException/HeadlessException.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,11 +34,12 @@ Screenshots
```
[the remark this answers is not in the document]
Comment (MaximMoinat) [reply README.md:34]: Same issue in [docs/Readme.md](https://github.com/OHDSI/WhiteRabbit/blob/master/docs/ReadMe.md) used for the Github.io page.

**Review by MaximMoinat — APPROVED** [review]
Looks good. I was able to build the distribution and test WhiteRabbit and RabbitInAHat applications. See inline comments for minor issues found.

It would be good to update the concept id hints before the v1.0 release.  This is now based on the vocabulary version from 21-DEC-20. I can open a separate PR for that.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (janblom) [reply README.md:34]: I will add this to my list of things to update before releasing.

File: README.md [hunk 2]
```diff
@@ -101,16 +102,23 @@ To generate the files ready for distribution, run `mvn install`.
 ### Testing
```
[the remark this answers is not in the document]
Comment (janblom) [reply README.md:102]: Good suggestion, I will fix this in a separate commit before releasing.

**Pull request merged**